datalad/distribution/siblings.py

```diff
@@ -239,30 +246,47 @@ def __call__(
         # at the top-level and vice versa
         worker = action_worker_map[action]
 
-        dataset = require_dataset(
-            dataset, check_installed=False, purpose='configure sibling')
-        refds_path = dataset.path
+        ds = require_dataset(
+            dataset,
+            # it makes no sense to use this command without a dataset
+            check_installed=True,
+            purpose='configure sibling')
+        refds_path = ds.path
 
         res_kwargs = dict(refds=refds_path, logger=lgr)
 
-        ds_name = op.basename(dataset.path)
+        ds_name = op.basename(ds.path)
 
         # do not form single list of datasets (with recursion results) to
         # give fastest possible response, for the precise of a long-all
         # function call
-        ds = dataset
-        for r in worker(
-                # always copy signature to below to avoid bugs!
-                ds, name,
-                ds.repo.get_remotes(),
-                # for top-level dataset there is no layout questions
-                _mangle_urls(url, ds_name),
-                _mangle_urls(pushurl, ds_name),
-                fetch, description,
-                as_common_datasrc, publish_depends, publish_by_default,
-                annex_wanted, annex_required, annex_group, annex_groupwanted,
-                inherit, get_annex_info,
-                **res_kwargs):
+
+        # minimize expensive calls to .repo
+        ds_repo = ds.repo
+
+        # prepare parameterization package for all worker calls
+        worker_kwargs = dict(
+            ds=ds,
+            repo=ds_repo,
+            name=name,
+            known_remotes=ds_repo.get_remotes(),
+            # for top-level dataset there is no layout questions
+            url=_mangle_urls(url, ds_name),
+            pushurl=_mangle_urls(pushurl, ds_name),
+            fetch=fetch,
+            description=description,
+            as_common_datasrc=as_common_datasrc,
+            publish_depends=publish_depends,
+            publish_by_default=publish_by_default,
+            annex_wanted=annex_wanted,
+            annex_required=annex_required,
+            annex_group=annex_group,
+            annex_groupwanted=annex_groupwanted,
+            inherit=inherit,
+            get_annex_info=get_annex_info,
+            res_kwargs=res_kwargs,
+        )
+        for r in worker(**worker_kwargs):
             yield r
```

Comment:
if all workers are generators (as probably they should be, but didn't check), here and below could be just

```suggestion
        yield from worker(**worker_kwargs)
```

Reply:
While adding "love" may be start adding type annotations? pandora's box of importing typing is already open -- so there should be no extra run time penalty, smth like
```
diff --git a/datalad/distribution/siblings.py b/datalad/distribution/siblings.py
index f3129f31f..9d4dd9dbd 100644
--- a/datalad/distribution/siblings.py
+++ b/datalad/distribution/siblings.py
@@ -16,6 +16,7 @@ import logging
 import os
 import os.path as op
 
+from typing import Iterator
 from urllib.parse import urlparse
 
 from datalad.interface.base import Interface
@@ -338,7 +339,7 @@ def _add_remote(
         as_common_datasrc, publish_depends, publish_by_default,
         annex_wanted, annex_required, annex_group, annex_groupwanted,
         inherit, get_annex_info,
-        **res_kwargs):
+        **res_kwargs) -> Iterator[dict]:
     # TODO: allow for no url if 'inherit' and deduce from the super ds
     #       create-sibling already does it -- generalize/use
     #  Actually we could even inherit/deduce name from the super by checking
```
for those workers, and that should (AFAIK) immediately give an idea if all workers are providing Iterators (mypy though seems to not alert if I change here to `yield from` and do not annotate all workers)

Reply:
I know too little about typing to be the one doing that, sorry. Quickly glancing at the docs, it also tells me:

> class typing.Iterator(Iterable[T_co])
A generic version of collections.abc.Iterator.
> Deprecated since version 3.9: collections.abc.Iterator now supports []. See PEP 585 and Generic Alias Type.

So this may not be the right path, who knows. Anyways, I do not see myself doing it in here.

Reply:
That being said, yes, `yield from` works, will change.